MySensors.h

```diff
@@ -307,6 +316,11 @@ MY_DEFAULT_RX_LED_PIN in your sketch instead to enable LEDs
 #include "drivers/AVR/DigitalIO/DigitalIO.h"
 #endif
 
+// SOFTSERIAL
+#if defined(MY_GSM_TX) != defined(MY_GSM_RX)
```

Comment:
Is it intentional that the error will fire if both flags are defined or both are not defined?

Reply:
hrm, I guess I'm missing the logic behind the statement. The goal was to fire an error if both are not the same (i.e. one is defined and the other is not)

Reply:
And testing it right now, it would seem to behave as I had hoped.

Reply:
Then I think what you want is something like
´#if (defined(MY_GSM_TX) != defined(MY_GSM_RX)) && defined(MY_GSM_TX)´

Reply:
Ok, well if it works as you like it, it looks fine by me. But I would assume that if you don't define either of them, and enter this segment, the error will fire.

Reply:
What is the benefit of the added ´&& defined(MY_GSM_TX)´ ?

I just ran the tests with all three conditions: both enabled, one enabled and both disabled, and the error message behaved as it should. I did however find that I need to add a separate check to make sure if softserial pins are not defined, then the hw serial must be available.

Reply:
I might be misunderstanding the conditions for the definitions,. but I would assume that it evaluates the expression in boolean manner, so the two defined() blocks either return true or false depending on the definition state, and the != comparison will evaluate to true if either of the states is different from the other. No matter if they are true or false.
The result should then be that the expression evaluates to true given that both define() blocks return true or false.

Reply:
I see now, the error of course kicks in if only one is set. If none or both are set, it is ok. My bad.

Reply:
All good :) Your comment made me run more extensive tests and discover the parallel problem of errors when not defining any Serial port for the modem to use. Got that sorted as well now :+1:

Reply:
Excellent.